src/main/java/org/betonquest/betonquest/quest/event/folder/FolderEvent.java

```diff
@@ -113,92 +165,29 @@ protected Void execute(@Nullable final Profile profile) throws QuestRuntimeExcep
         } else {
             chosenList.addAll(Arrays.asList(events));
         }
-
-        final Long execDelay = getInTicks(delay, profile);
-        final Long execPeriod = getInTicks(period, profile);
-
-        if (execDelay == null && execPeriod == null) {
-            for (final EventID event : chosenList) {
-                BetonQuest.event(profile, event);
-            }
-        } else if (execPeriod == null) {
-            final FolderEventCanceller eventCanceller = createFolderEventCanceller(profile);
-            callSameSyncAsyncContext(new BukkitRunnable() {
-                @Override
-                public void run() {
-                    eventCanceller.destroy();
-                    if (eventCanceller.isCancelled()) {
-                        return;
-                    }
-                    for (final EventID event : chosenList) {
-                        BetonQuest.event(profile, event);
-                    }
-                }
-            }, execDelay, -1);
-        } else {
-            if (execDelay == null && !chosenList.isEmpty()) {
-                final EventID event = chosenList.removeFirst();
-                BetonQuest.event(profile, event);
-            }
-            if (!chosenList.isEmpty()) {
-                final FolderEventCanceller eventCanceller = createFolderEventCanceller(profile);
-                callSameSyncAsyncContext(new BukkitRunnable() {
-                    @Override
-                    public void run() {
-                        final EventID event = chosenList.pollFirst();
-                        if (eventCanceller.isCancelled() || event == null) {
-                            eventCanceller.destroy();
-                            this.cancel();
-                            return;
-                        }
-                        BetonQuest.event(profile, event);
-                    }
-                }, execDelay == null ? execPeriod : execDelay, execPeriod);
-            }
-        }
-        return null;
+        return chosenList;
     }
 
     private FolderEventCanceller createFolderEventCanceller(@Nullable final Profile profile) {
         if (cancelOnLogout && profile != null) {
-            return new QuitListener(log, profile);
+            return new QuitListener(betonQuest, betonQuest.getLoggerFactory().create(this.getClass()), profile);
```

Comment:
A class never create his own logger, it gets always injected from outside. So please let the logger be injected from the class initializing this class.